src/lustre/lustreSyntaxChecks.ml

```diff
@@ -612,6 +658,17 @@ and check_struct_items ctx items =
   | (ArraySliceStructItem (pos, _, _)) :: _
     -> syntax_error pos UnsupportedAssignment
 
+(* Within a frame block, make sure vars in the LHS of 'ni' appear somehwere in 'vars'  *)
+and check_frame_vars pos vars ni = 
+  let vars_of_ni = List.map snd (LAH.vars_lhs_of_eqn_with_pos ni) in
+  (* 'find_vars' stores the original variables from 'vars_of_ni' and their corresponding
+     variables found in 'vars', if they exist *)
+  let find_vars = List.map (fun var1 -> (var1, (List.find_opt (fun var2 -> var2 = var1) vars))) vars_of_ni in
+  let no_match = List.filter (fun x -> match x with | (_, None) -> true | (_, Some _) -> false) find_vars in
+  match no_match with
+    | [] -> Res.ok ()
+    | (var, _) :: _ -> syntax_error pos (MisplacedVarInFrameBlock var)
```

Comment:
This could be more concisely expressed with set operations:
```
let open HString in
let unlisted =
  HStringSet.diff (HStringSet.of_list vars_of_ni) (HStringSet.of_list vars)
in
match HStringSet.choose_opt unlisted with
| None -> Res.ok ()
| Some var -> syntax_error pos (MisplacedVarInFrameBlock var)
```